pareto/utilities/cm_utils/opt_utils.py

```diff
@@ -0,0 +1,196 @@
+#####################################################################################################
+# PARETO was produced under the DOE Produced Water Application for Beneficial Reuse Environmental
+# Impact and Treatment Optimization (PARETO), and is copyright (c) 2021-2024 by the software owners:
+# The Regents of the University of California, through Lawrence Berkeley National Laboratory, et al.
+# All rights reserved.
+#
+# NOTICE. This Software was developed under funding from the U.S. Department of Energy and the U.S.
+# Government consequently retains certain rights. As such, the U.S. Government has been granted for
+# itself and others acting on its behalf a paid-up, nonexclusive, irrevocable, worldwide license in
+# the Software to reproduce, distribute copies to the public, prepare derivative works, and perform
+# publicly and display publicly, and to permit others to do so.
+#####################################################################################################
+"""
+Functions used in run_infrastructure_analysis.py
+"""
+
+from pareto.other_models.CM_module.models.qcp_br import build_qcp_br
+import pyomo.environ as pyo
+from pareto.utilities.cm_utils.gen_utils import obj_fix
+
+
+def max_theoretical_recovery_flow_opt(
+    model, treatment_unit, desired_li_conc, tee=False
+):
+    """This function computes the largest flow possible to
+    the treatment unit while still keeping the Li
+    concentration above the desired level.
+
+    This function ignores all infrastructure - this is only
+    based on mass balance along.
```

Comment:
I'm not sure what "mass balance along" means - I think this needs to be reworded.